module-03-react/src/components/Gallery.jsx

```diff
@@ -0,0 +1,146 @@
+import React, {
+    useEffect,
+    useState,
+    useImperativeHandle,
+    forwardRef,
+} from "react";
+import axios from "axios";
+import "/src/styles/components.css";
+import Card from "./Card.jsx";
+
+const baseURL = "https://swapi.dev/api/";
+
+const Gallery = forwardRef(
+    ({ setImg, search, setSearch, url, setOpenModal, setChoosedCard }, ref) => {
+        const [cards, setCards] = useState([]);
+        const [allResults, setAllResults] = useState([]);
+        const [page, setPage] = useState(1);
+
+        const [loading, setLoading] = useState(false);
+        const [error, setError] = useState(null);
+
+        const bottomButton = document.querySelector(".bottom__btn");
+        let cardCounter = 0;
+
+        useEffect(() => {
+            const bottomButton = document.querySelector(".bottom__btn");
+
+            const getAPI = async () => {
+                setLoading(true);
+                setError(null);
+
+                let allResults = [];
+                let nextPage = baseURL + url;
+                try {
+                    while (nextPage) {
+                        try {
+                            const response = await axios.get(nextPage);
+                            const { results, next } = response.data;
+                            allResults = [...allResults, ...results];
+                            nextPage = next;
+                        } catch (error) {
+                            nextPage = null;
+                        }
+                    }
+                    setAllResults(allResults);
+                    setCards(allResults.slice(0, 15));
+                    if (allResults.length <= 15 && bottomButton) {
+                        bottomButton.classList.add("is-hidden");
```

Comment:
Direct DOM manipulation with `bottomButton.classList.add("is-hidden")` breaks React's declarative paradigm. Instead, control visibility through component state and conditional rendering. Consider using the `clsx` library to manage class names conditionally:

```javascript
import clsx from 'clsx';

// Add state to track visibility
const [isButtonHidden, setIsButtonHidden] = useState(allResults.length <= 15);

// Then in JSX:
// <button className={clsx('bottom__btn', { 'is-hidden': isButtonHidden })}>Load More</button>
```

This approach keeps styling concerns within React's state management, making your code more predictable, maintainable, and aligned with React's component model